### PR TITLE
docs(agent): add more explanations about tracetest agent

### PR DESCRIPTION
**Pull request by adnanrahic**
This PR adds a better explanation about Tracetest Agent, including a concepts page.

## Changes

-

## Fixes

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Loom video

Add your loom video here if your work can be visualized

### CODE REVIEW

File: docs/docs/concepts/agent.mdx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+keywords:
+  - tracetest
+  - trace-based testing
+  - observability
+  - distributed tracing
+  - testing
+image: https://res.cloudinary.com/djwdcmwdz/image/upload/v1689693872/docs/Blog_Thumbnail_28_ugy2yy.png
+---
+
+Tracetest Agent is a lightweight, dependency-free, agent that runs locally in your development environment, or as a Docker container in your Cloud Native infrastructure.
```
Comment (jfermi) [inline docs/docs/concepts/agent.mdx:14]:
```suggestion
Tracetest Agent is a lightweight, dependency-free agent that runs locally in your development environment, or as a Docker container in your Cloud Native infrastructure.
```

File: docs/docs/concepts/agent.mdx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+## How it works
```
Comment (jfermi) [inline docs/docs/concepts/agent.mdx:16]:
```suggestion
## How It Works
```

File: docs/docs/configuration/agent.mdx [hunk 3]
```diff
@@ -13,12 +13,20 @@ image: https://res.cloudinary.com/djwdcmwdz/image/upload/v1689693872/docs/Blog_T
...
+[Start Tracetest Agent](../getting-started/installation.mdx) in your environment to begin testing!
 
-Or, [sign in to Tracetest](https://app.tracetest.io/) and follow the instructions on the Agent Settings page.
+Or, [sign in to Tracetest](https://app.tracetest.io/) and follow the instructions on the `Settings > Agent` page.
 
 ![](https://res.cloudinary.com/djwdcmwdz/image/upload/v1696941536/docs/app.tracetest.io_organizations_ttorg_e66318ba6544b856_environments_ttenv_0e807879e2e38d28_settings_j4zt1b.png)
+
+## Configure Tracetest Agent
+
+Once started, Tracetest Agent exposes OTLP ports `4317` and `4318` to ingest traces via `gRPC` and `HTTP` and is responsible for running tests. Read the detailed concepts, [here](../concepts/agent).
```
Comment (jfermi) [inline docs/docs/configuration/agent.mdx:28]:
```suggestion
Once started, Tracetest Agent exposes OTLP ports `4317` and `4318` to ingest traces via `gRPC` and `HTTP` and is responsible for running tests. Read the detailed concepts [here](../concepts/agent).
```

**Review by jfermi — COMMENTED** [review]
A few editing suggestions.

File: docs/docs/concepts/agent.mdx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  - testing
+image: https://res.cloudinary.com/djwdcmwdz/image/upload/v1689693872/docs/Blog_Thumbnail_28_ugy2yy.png
+---
+
+Tracetest Agent is a lightweight, dependency-free agent that runs locally in your development environment, or as a Docker container in your Cloud Native infrastructure.
+
+## How It Works
+
+- Tracetest Agent listens for new tests you run from [Tracetest](https://app.tracetest.io). It executes tests in the environment where it is running and returns the response from the trigger. The environment can be either `localhost`, Docker, or Kubernetes.
+- Tracetest Agent listens on OTLP ports `4317` (gRPC) and `4318` (HTTP) for OpenTelemetry trace spans to capture distributed trace data and relay it to [Tracetest](https://app.tracetest.io).
```
Comment (mathnogueira) [inline docs/docs/concepts/agent.mdx:19]: Is it worth to mention that the agent is able to connect to your datastore that is not accessible outside your environment. E.g. a jarger instance running in the cluster without an ingress controller

**Pull request merged**